modify site to fetch json

poncem91 opened the pull request:
Commit modifies DataServlet java file to return JSON data, modifies script JS file to fetch the JSON data and present it when sayHello() function is called.

poncem91 committed:
modify site to fetch json

Commit modifies DataServlet java file to return JSON data, modifies script JS file to fetch the JSON data and present it when sayHello() function is called.

peap commented on portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:
```diff
@@ -13,20 +13,34 @@
 @WebServlet("/data")
 public class DataServlet extends HttpServlet {
+    
+    private ArrayList<String> helloMessages;
```
nit: check the indentation here

poncem91 replied:
Will fix this, thanks for pointing this out!

peap commented on portfolio/src/main/webapp/script.js:
```diff
@@ -119,7 +119,13 @@ document.addEventListener('keydown', function(event) {
 function sayHello() {
-  fetch('/data').then(response => response.text()).then((helloMessage) => {
-    document.getElementById('hello-container').innerText = helloMessage;
+  fetch('/data').then(response => response.json()).then((helloMessage) => {
```
I found a bug :) 

![image](https://user-images.githubusercontent.com/2783959/83467064-bc6d8880-a42d-11ea-86a4-81e9cb412798.png)

poncem91 replied:
Thanks for catching that! Definitely not what I intended. Will fix this

poncem91 replied:
Done! :)

peap replied:
The `Hello!` button is gone from the Mafe_jsonWalkthrough branch, and I don't see it in master, either. It's still hanging around in Mafe_FetchWalkthrough, though. Maybe it got lost in some merge commit?

poncem91 replied:
I tried merging one of the previous branches and it did some weird things... I just reverted some of it and will be making a new pull request for the Mafe_jsonWalkthrough branch (the button should be there now) along with a pull request for a branch with the POST form. I’m hoping this brings back to where they should be

poncem91 committed:
fix welcome message bug and other nits

Commit fixed bug in welcome message implementation by fixing a typo that was responsible for the bug.
Deletes no longer necessary TODO comment and fixes indent on a line.
